### ISSUE TITLE
sf::Texture::getMaximumSize() segfaults

### ISSUE THREAD

**Issue by dogunbound** [SFML/SFML#2602]
## sf::Texture::getMaximumSize() segfaults

I think it has to deal with loading up the opengl interface. If you load a font, texture, or something that uses the graphics card, then this issue is not relevant. 

## Your environment

* ArchLinux
*SFML 2.6.0
* GCC 13.1.1
* N/A

## Steps to reproduce

Put this in main.cpp and compile and run
```cpp
#include <iostream>
#include "SFML/Graphics/Texture.hpp"
int main() {
  std::cout << sf::Texture::getMaximumSize();
}
```

## Expected behavior

```
[dogunbound@archlinux cmake-sfml-project-master]$ ./build.sh ; build/CMakeSFMLProject 
-- Found SFML 2.6.0 in /usr/local/lib/cmake/SFML
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /home/dogunbound/SFML/cmake-sfml-project-master/build
[ 50%] Building CXX object CMakeFiles/CMakeSFMLProject.dir/src/main.cpp.o
[100%] Linking CXX executable CMakeSFMLProject
[100%] Built target CMakeSFMLProject
16384
```

## Actual behavior

```
[dogunbound@archlinux cmake-sfml-project-master]$ ./build.sh ; build/CMakeSFMLProject 
-- Found SFML 2.6.0 in /usr/local/lib/cmake/SFML
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /home/dogunbound/SFML/cmake-sfml-project-master/build
[ 50%] Building CXX object CMakeFiles/CMakeSFMLProject.dir/src/main.cpp.o
[100%] Linking CXX executable CMakeSFMLProject
[100%] Built target CMakeSFMLProject
Segmentation fault (core dumped)
```

**Comment by ChrisThrasher** [comment]
Sounds like something @binary1248 might have some insight about. Some Discord users are saying that the crash goes away if you first create a render window so this appears to be related to global GL initialization.

**Comment by Bambo-Borris** [comment]
Reproduced this on the following environments & configurations: 

Windows with 2.6.0 tagged release, 2.6.x branch and master branch (using CMake template project)
Ubuntu via WSL on 2.6.0 tagged release and master branch

**Comment by dogunbound** [comment]
SFML v3 (master) also crashes.

**Comment by eXpl0it3r** [comment]
Fixed with #2603, thanks for the report! 🙂

**Comment by ChrisThrasher** [comment]
<img width="305" alt="Screenshot 2023-07-04 at 3 56 58 PM" src="https://github.com/SFML/SFML/assets/39244355/1e6818ef-f057-4806-96e7-11324a3d81ac">

Quick turnaround! Great work, everybody!
